modules/snooker/src/main/ts/ephox/snooker/api/TableOperations.ts

```diff
@@ -359,9 +359,14 @@ const firstColumnIsLocked = (_warehouse: Warehouse, details: Structs.DetailExt[]
 const lastColumnIsLocked = (warehouse: Warehouse, details: Structs.DetailExt[]) =>
   Arr.exists(details, (detail) => detail.column + detail.colspan >= warehouse.grid.columns && detail.isLocked);
 
-const getColumnsWidth = (details: Structs.DetailExt[]) => {
+const getColumnsWidth = (warehouse: Warehouse, details: Structs.DetailExt[]) => {
+  const columns = Blocks.columns(warehouse);
   const uniqueCols = ColUtils.uniqueColumns(details);
-  return Arr.foldl(uniqueCols, (acc, detail) => acc + Width.getOuter(detail.element), 0);
+  return Arr.foldl(uniqueCols, (acc, detail) => {
```

Comment:
This was the primary other cause, as it used the "unique" cols width. However, the cell used in the `Width.getOuter` call was the one with the colspan.

So if we had 3 cols, each ~33% and we included the colspan cell in the selection, then the delta would be `66% + 33%` (ie all three columns). However, it should have only been the 2 columns represented by the colspan (eg `33% + 33%`).

As such, this change just makes it use the column index from `uniqueCols` and then uses `Blocks.columns` to find a cell that appropriately represents the column in the table to get the outer width.